qrcode_app/lib/screens/home_screen.dart

```diff
@@ -0,0 +1,138 @@
+import 'package:barcode_scan2/barcode_scan2.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:qr_flutter/qr_flutter.dart';
+import 'package:qrcode_app/widgets/expandable_fab.dart';
+import 'package:qrcode_app/presentation/seed_presenter.dart';
+// import 'package:scan/scan.dart';
+
+class HomeScreen extends StatefulWidget {
+  const HomeScreen({Key? key, required this.presenter}) : super(key: key);
+  final SeedPresenter presenter;
+  @override
+  _HomeScreenState createState() => _HomeScreenState();
+}
+
+class _HomeScreenState extends State<HomeScreen> {
+  ScanResult? scanResult;
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Home'),
+      ),
+      body: Center(
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.center,
+          mainAxisAlignment: MainAxisAlignment.spaceAround,
+          children: [
+            StreamBuilder<PresenterState>(
+                stream: widget.presenter.streamData,
+                builder: (context, snapshot) {
+                  if (!snapshot.hasData)
+                    return Container();
+                  else if (snapshot.data is LoadingState)
+                    return Center(
+                      child: CircularProgressIndicator(),
+                    );
+                  else if (snapshot.data is FailureState)
+                    return Container(
+                      child: Text('Error'),
+                    );
+                  else {
+                    print(snapshot.data!.seed.seed);
+                    return CustomPaint(
+                      size: Size.square(200),
+                      painter: QrPainter(
+                        data: snapshot.data!.seed.seed,
+                        version: QrVersions.auto,
+                        eyeStyle: const QrEyeStyle(
+                          eyeShape: QrEyeShape.square,
+                          color: Color(0xff128760),
+                        ),
+                        dataModuleStyle: const QrDataModuleStyle(
+                          dataModuleShape: QrDataModuleShape.circle,
+                          color: Color(0xff1a5441),
+                        ),
+                      ),
+                    );
+                  }
+                }),
+            StreamBuilder<int>(
+                stream: widget.presenter.expirationData,
```

Comment:
**question:** I'm noticing that when I tap generate, the QR Code displays about a full second ahead of when the expiration information displays. Any ideas what's causing that, and how we could get both to appear at the same time?